checks.d/riakcs.py

```diff
@@ -107,8 +123,191 @@ def _get_stats(self, s3, aggregation_key):
 
         return stats
 
-
-    # We need this as the riak cs stats page returns json with duplicate keys
     @classmethod
     def load_json(cls, text):
-        return json.JSONDecoder(object_pairs_hook=multidict).decode(text)
+        data = json.loads(text)
+        if "legend" in data:
+            # riak cs before v2.1 had duplicate keys
+            data = json.JSONDecoder(object_pairs_hook=multidict).decode(text)
+        return data
+
+
+STATS_METHODS = {
+    "one": "count",
+}
+
+# This list includes most S3 API metrics as well as memory stats. Some
+# have been excluded, mainly just to keep size of the default set of
+# metrics somewhat reasonable.
+#
+# Excluded S3 metrics:
+#   - bucket_acl_(get|put)
+#   - object_acl_(get|put)
+#   - bucket_policy_(get|put|delete)
+#   - *_in_(one|total)
+#   - *_time_error_*
+#   - *_time_100
+#
+# Any of these excluded metrics in addition to many others (there are
+# over 1000 to choose from) can be added by specifying them in the
+# riakcs.yaml config file under the "metrics" key in the instance
+# config; the value should be a list of metric names.
+#
+# Helpful references:
+# - https://github.com/basho/riak_cs/wiki/Riak-cs-and-stanchion-metrics
```

Comment:
I chose to only collect a limited set of S3 API and memory metrics by default. I'm not sure how to identify the set that most people will want, so I just went with a set that seemed reasonable for my usage (although I haven't done anything with it yet, so this may evolve). Here are some reasons for each of the exclusions listed above:

- `bucket_acl_(get|put)` and `object_acl_(get|put)` - I'm not using ACLs.
- `bucket_policy_(get|put|delete)` - I'm not doing any automated bucket policy updates.
- `*_in_(one|total)` - guessing this is a similar metric to the `_out_` variants.
- `*_time_error_*` - it seems like the timing of errors is less important than the count of errors.
- `*_time_100` - had a hard time figuring out what this even means. Is it 99.9th percentile? Decided to exclude it since it was not in the old metrics format.